core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -182,6 +187,17 @@ object Invoker {
 
     val invokerInstance = InstanceId(assignedInvokerId)
     val msgProvider = SpiLoader.get[MessagingProvider]
+    if (!msgProvider.ensureTopic(
```

Comment:
should this happen inside the kafka messaging provider since this may not be applicable to an alternate bus implementation?

Reply:
The keys of the Map are pretty kafka specific.  The values are both kafka specific and specific to the topic being created (see similar block in the controller where we read different fields of WhiskConfig).

I think using a Map[String,String] is probably workable in a generic SPI, but when populating the map you end up needing to know the SPI implementation you are talking to.

An alternative would be to not pass a Map and instead pass just the config object and the topic name. We could look at the topic prefix inside the ensureTopic implementation of the kafka messaging provider and use it to figure out which fields of the config to extract.  

I think this really doesn't help make it more generic, since the details of the messaging provider are still being leaked into the WhiskConfig.  I could make the change though without much effort.

Reply:
The general question is (which maybe has been answered and I just don't know it), how are SPI implementations supposed to plumb their configuration values from the deploying environment through to the implementation code that needs them.  Adding them all to WhiskConfig seems wrong, but I don't know what else to do.

Reply:
@markusthoemmes @tysonnorris I think we’ve discussed this in the past. I have two comments: whisk config will get replaced and so the maps will likely be gone. I’ll link a PR as an example. Alternatively the Plugins would provide a requiredProperties method which get added to the whisk config object. It’s ok with me to defer this until to a separate pr down the line.

Reply:
Using typesafe config or pureconfig, I think, will make this issue moot since there will be no single "universal" config class; rather SPI impls can declare their own etc, and only that impl will reference its config. 

The only questions is whether there should be a universal config validation, which SPI impls should be able to contribute to? This I think can be added later, if at all. In the meantime, the difference is that "missing configs" required by SPI impls would cause failures ONLY when that impl is loaded. In all cases, impls should be able to require different config keys - but I don't think that is being questioned.

Reply:
Once we have pureconfig and I can get the kafka-specific params internal to the SPI, I expect this block of code will reduce to

```
if (!msgProvider.ensureTopic(config,  "invoker" + assignedInvokerId) {
      abort(s"failure during msgProvider.ensureTopic for topic invoker$assignedInvokerId")
 }
```